### PR TITLE
Enable agent path template customization for azure_msi node attestor plugin

### LINKED ISSUE #3420: Allow setting Agent Path Template for Azure MSI Node Attestation
I believe the Azure integration should allow the same customizations for Agent ID format that GCP, AWS, x509pop, and sshpop provide.

Owners of SPIRE operations that deploy across several different provider types are unable to keep a consistent naming scheming for their Agents, adding complexity to any maintenance procedures and service/registrant integrations which rely upon Agent ID format.

Example for GCP integration: https://github.com/spiffe/spire/blob/main/pkg/server/plugin/nodeattestor/gcpiit/iit.go#L16

However for Azure integration, there is no such configuration: https://github.com/spiffe/spire/blob/c13d763178c35dde4be790f5e9625121e4ca30ed/doc/plugin_server_nodeattestor_azure_msi.md
```
spiffe://<trust domain>/spire/agent/azure_msi/<tenant_id>/<principal_id>
```

Note this doesn't necessarily fully conflict with #2712, depending on resolution of keeping `plugin_type` within the ID, but it is certainly related.

### PR DESCRIPTION
**Pull request by guilhermocc**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**
- Added the ability to use a custom agent path template for azure_msi server node attestor plugin.  

**Description of change**
- Keep the actual azure_msi agent SPIFFE ID form as the default to maintain compatibility
- azure_msi plugin now accepts a new optional configuration `agent_path_template` that is used for building the agent's SPIFFE ID.
- Added missing documentation for gcp_iit plugin

**Which issue this PR fixes**
fixes #3420

### CODE REVIEW

**Comment by azdagron** [comment]
Thanks for opening this, @guilhermocc ! I'll see if I can get it reviewed here shortly. Just for information, you don't need to bother keeping the branch up to date. Once it has been approved, we (the maintainers) will update it as needed.

File: pkg/common/plugin/azure/msi.go
Comment (azdagron) [inline pkg/common/plugin/azure/msi.go:45]: I did a double take and had to remind myself what happens with the JSON decoder when there are two fields with the same tag. Fortunately the top level field takes precedence over the embedded struct but this is going to be a little subtle if there is any code that tries to use the `.Subject` field (since it will be unset). Should be easy enough to catch in testing though. I think we're ok here.

Comment (guilhermocc) [reply pkg/common/plugin/azure/msi.go:45]: I thought about creating a new struct for only storing claims/fields that we want to expose and use. But we would still have another struct being composed by  jwt.Claims only, since we use the `ValidateWithLeeway` method for validation, but I'm not sure if that would make the code clearer since we will be going to deserialize the token two times for two different structs, what do you think?

Comment (azdagron) [reply pkg/common/plugin/azure/msi.go:45]: I think we're ok with it as-is. If the `Subject` field is used at all the unit-tests should catch that it is empty.

**Review by azdagron — DISMISSED** [review]

File: pkg/common/plugin/azure/msi_test.go
Comment (MarcosDY) [inline pkg/common/plugin/azure/msi_test.go:156]:
```suggestion
			td, _ := spiffeid.RequireTrustDomainFromString(test.args.td)
```

File: pkg/common/plugin/azure/msi_test.go
Comment (MarcosDY) [inline pkg/common/plugin/azure/msi_test.go:127]: can you set far names in all cases you create a new struct? it can resutl in unexpected errors if we choose to add more fields

File: pkg/common/plugin/azure/msi_test.go
Comment (MarcosDY) [inline pkg/common/plugin/azure/msi_test.go:122]: instead of want error, can you instead validate the error you are getting? it is possible to get into situations where you get another error and not the expected one

File: pkg/server/plugin/nodeattestor/azuremsi/msi_test.go
Comment (MarcosDY) [inline pkg/server/plugin/nodeattestor/azuremsi/msi_test.go:246]:
```suggestion
	selectorValues := append([]string(nil), vmSelectors...)
```

File: doc/plugin_server_nodeattestor_azure_msi.md
Comment (guilhermocc) [reply doc/plugin_server_nodeattestor_azure_msi.md:23]: Sure!

**Review by MarcosDY — APPROVED** [review]
LGTM!!!

**Pull request merged**
